pkg/attestlib/pkix.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package attestlib
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"github.com/pkg/errors"
+)
+
+type pkixSigner struct {
+	PrivateKey         interface{}
+	PublicKeyID        string
+	SignatureAlgorithm SignatureAlgorithm
+}
+
+// NewPkixSigner creates a Signer interface for PKIX Attestations. `privateKey`
+// contains the PEM-encoded private key. `publicKeyID` is the ID of the public
+// key that can verify the Attestation signature.
+func NewPkixSigner(privateKey []byte, publicKeyID string, alg SignatureAlgorithm) (Signer, error) {
```

Comment:
Move publicKeyID to be the last parameter, and add a  comment saying that in almost all cases it should be left empty , and then the key ID will be generated automatically

Reply:
done.